### PR TITLE
Fixed compile error at using SQLSyntaxSupportFactory

### PR DESCRIPTION
**Pull request by gakuzzzz**
SQLSyntaxProvider is package private.

### CODE REVIEW

File: scalikejdbc-syntax-support-macro/src/test/scala/scalikejdbc/AutoSpec.scala [hunk 1]
```diff
@@ -1,5 +1,6 @@
-package scalikejdbc
+package foo
```
Comment (seratch) [inline scalikejdbc-syntax-support-macro/src/test/scala/scalikejdbc/AutoSpec.scala]: Move AutoSpec.scala to src/test/scala/foo

Comment (gakuzzzz) [inline scalikejdbc-syntax-support-macro/src/test/scala/scalikejdbc/AutoSpec.scala]: Alrgiht.
DBSettings too?

Comment (seratch) [inline scalikejdbc-syntax-support-macro/src/test/scala/scalikejdbc/AutoSpec.scala]: Ah, no. I just meant `foo.AutoSpec` should be there if you change the package.

Comment (gakuzzzz) [inline scalikejdbc-syntax-support-macro/src/test/scala/scalikejdbc/AutoSpec.scala]: ya

**Comment by xuwei-k** [comment]
:+1:

**Pull request merged**